src/ol/interaction/Snap.js

```diff
@@ -263,12 +291,20 @@ class Snap extends PointerInteraction {
   /**
    * @param {import("../MapBrowserEvent.js").default} evt Map browser event.
    * @return {boolean} `false` to stop event propagation.
+   * @api
    */
   handleEvent(evt) {
     const result = this.snapTo(evt.pixel, evt.coordinate, evt.map);
     if (result) {
       evt.coordinate = result.vertex.slice(0, 2);
       evt.pixel = result.vertexPixel;
+      this.dispatchEvent(
+        new SnapEvent(SnapEventType.SNAP, {
+          vertex: result.vertex.slice(0, 2),
+          vertexPixel: result.vertexPixel,
```

Comment:
If you'd assign `evt.coordinate` and `evt.pixel` here instead, users would be able to modify the snap coordinate, which might be useful.

Reply:
If you decide not to change that, then also create a copy of the pixel so users can neither modify `coordinate` nor `pixel`.

Reply:
Changed to assign evt.coordinate and evt.pixel as suggested. 
Curious what is the example use case for the users wanting to modify snap coordinate?